caproto/_data.py

```diff
@@ -1041,6 +1068,8 @@ def preprocess_value(self, value):
                 value = b''
             elif len(value) == 1:
                 value = value[0]
+            elif self.is_compatible_array(value):
+                value = value.tobytes()
```

Comment:
This costs a memory copy. I wonder if memoryview would work.

Reply:
Goal of `preprocess_value` as I recall is to make it a consistent `bytes` string for the `putter` handler to accept/reject/modify. 

I suppose if the client code is _writing_ such arrays directly, then it may be assumed that their handlers can accept such arrays and won't presume a `bytes`-like interface. I'm not sure there's a 100% correct answer here.

The alternative would be just allowing the array through unchanged. Storing such a mutable value could lead to problems unless we copy it down the line (another thing I'd have to double-check...).

Reply:
I swayed your opinion, and now I'll try to sway it back.

`.tobytes()` ends up being too slow for large arrays, but we _can_ copy it and still maintain a reasonably fast update speed.
The array interface provides an option to check if the array is flagged as read-only. The read-only nature is only a matter of user-intent and isn't _actually_ read-only, of course. 

How do you feel about the recently pushed commits?